### PR TITLE
Feature: Code Editor and new design for lightning menu

### PR DESCRIPTION
**Pull request by devrk96**

### CODE REVIEW

File: app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx [hunk 1]
```diff
@@ -75,16 +89,29 @@ export const LightningMenuTrigger = (props: LightningMenuTriggerProps) => {
...
+          autoFocus={false}
+          hoverOpenDelay={1000}
+          content={LIGHTNING_MENU_DATA_TOOLTIP}
+          openOnTargetFocus={false}
+        >
+          <IconWrapper {...iconProps}>
+            <LightningIcon />
+          </IconWrapper>
+        </BlueprintTooltip>
+      )}
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx:114]: No need for fallback here. Move the the new design for lighting menu

File: app/client/src/components/editorComponents/CodeEditor/BindingPrompt.tsx [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+  color: #858282;
+  box-shadow: 0px 12px 34px -6px rgba(0, 0, 0, 0.75);
   border-radius: 2px;
-  background-color: ${Colors.BLUE_CHARCOAL};
+  background-color: ${(props) =>
+    props.editorTheme === EditorTheme.DARK
+      ? "#262626"
+      : props.editorTheme === EditorTheme.LIGHT
+      ? "#E8E8E8"
+      : Colors.BLUE_CHARCOAL};
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/BindingPrompt.tsx:21]: Can we have named colors or get this from the theme?

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts [hunk 3]
```diff
@@ -168,6 +163,40 @@ export const EditorWrapper = styled.div<{
...
+              ? "#F7F7F7"
+              : props.isFocused
+              ? "#E8E8E8"
+              : "#FFFFFF"
+          };
+        }
+      }
+    }
+    `
+      : null}
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:199]: Why have this fallback here?

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts [hunk 4]
```diff
@@ -314,8 +355,8 @@ export const DynamicAutocompleteInputWrapper = styled.div<{
     .lightning-menu {
       background: ${(props) =>
         !props.isNotHover
-          ? props.skin === Skin.DARK
-            ? Colors.ALABASTER
+          ? props.skin === Skin.DARK || props.skin === Skin.LIGHT
+            ? "#716E6E"
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:359]: Dont keep fallback

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts [hunk 5]
```diff
@@ -331,4 +372,14 @@ export const DynamicAutocompleteInputWrapper = styled.div<{
...
+  ${(props) =>
+    props.skin === Skin.DARK || props.skin === Skin.LIGHT
+      ? `
+        border: 0px;
+        border-radius: 0px;
+        .lightning-menu {
+          z-index: 1 !important;
+        }
+      `
+      : null};
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:384]: Dont keep fallback

File: app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx [hunk 6]
```diff
@@ -19,6 +23,15 @@ const LightningIconWrapper = styled.span<{
...
   border-radius: 2px;
+  ${(props) =>
+    props.skin === Skin.DARK || props.skin === Skin.LIGHT
+      ? `
+      border-radius: 0px;
+      svg path {
+        fill: ${!props.isFocused && props.color};
+      }
+      `
+      : null};
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx:34]: Dont keep fallback

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx (hunk 1 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx]: Resolved

File: app/client/src/components/editorComponents/CodeEditor/BindingPrompt.tsx (hunk 2 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/CodeEditor/BindingPrompt.tsx]: Yes, added the named colors here.

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts (hunk 3 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/CodeEditor/styledComponents.ts]: Removed this fallback.

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts (hunk 4 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/CodeEditor/styledComponents.ts]: Resolved.

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts (hunk 5 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/CodeEditor/styledComponents.ts]: Resolved.

File: app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx (hunk 6 above)
Comment (devrk96) [reply app/client/src/components/editorComponents/LightningMenu/LightningMenuTrigger.tsx]: Resolved.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**